### PR TITLE
Monitor docker peers

### PR DESCRIPTION
**Pull request by raboof**
Preliminary code for monitoring connections from different docker instances.

Main missing feature is including some way to disambiguate connections from different docker hosts. Options there would be to include a docker host id, or docker container id's.

What'd work best depends on what kind of information we can get from other sources - making as few assumptions as possible would probably mean including the docker container id for each connection.

### CODE REVIEW

**Comment by raboof** [comment]
The `local_ips` collection contains the local IP's of all containers combined. We might want to use a separate set per docker container.

File: sockets/proc_net_tcp/proc_net_tcp.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func hexToIP(word string, ipv6 bool) (net.IP, error) {
 }
 
 // Parses the /proc/net/tcp file
-func parseProcNetTCP(input io.Reader, ipv6 bool) ([]*sockets.SocketInfo, error) {
+func ParseProcNetTCP(input io.Reader, ipv6 bool, containerId string) ([]*sockets.SocketInfo, error) {
```
Comment (whazenberg) [inline sockets/proc_net_tcp/proc_net_tcp.go]: What's your convention on starting funcs with a capital?

File: beater/connections.go [hunk 2]
```diff
@@ -45,8 +59,10 @@ func getSocketInfoFromTcpDiag(pollInterval time.Duration, socketInfo chan<- *soc
 	}
 }
 
-func getSocketInfo(enableTcpDiag bool, pollInterval time.Duration, socketInfo chan<- *sockets.SocketInfo) {
-	if enableTcpDiag {
+func getSocketInfo(enableDocker, enableTcpDiag bool, pollInterval time.Duration, socketInfo chan<- *sockets.SocketInfo) {
+	if enableDocker {
+		getSocketInfoFromDocker("unix:///var/run/docker.sock", pollInterval, socketInfo)
```
Comment (raboof) [inline beater/connections.go]: I don't think the docker socket is ever somewhere else, but in the future we might want to allow connecting over TCP+TLS. For that to work we'd also need a way to configure the right certificates etc, though.

File: sockets/proc_net_tcp/proc_net_tcp.go (hunk 1 above)
Comment (raboof) [inline sockets/proc_net_tcp/proc_net_tcp.go]: In Go, public methods start with a capital, private ones with a lowercase letter

Comment (whazenberg) [inline sockets/proc_net_tcp/proc_net_tcp.go]: Good to know.

**Pull request merged**